astroid/interpreter/_import/spec.py

```diff
@@ -291,9 +291,10 @@ def _precache_zipimporters(path=None):
 def _search_zip(modpath, pic):
     for filepath, importer in list(pic.items()):
         if importer is not None:
-            found = importer.find_module(modpath[0])
+            found = getattr(importer, "find_spec", importer.find_module)(modpath[0])
```

Comment:
Why not simply replacing `find_module` with `find_spec`?

Reply:
I tried that first but the tests were failing - it seems that the `importer` object that appeared here did not have the `find_spec` attribute, I'm not sure why. This way the change is backwards compatible because it will simply fall back to `find_module` if `find_spec` doesn't exist, so I think it's anyways a safer choice.

Reply:
@GergelyKalmar i investigated a bit the need for `find_spec` or `find_module` methods.
In fact the `pic.items()` container is made of `FileFinder` objects and `ZipFinder`objects. While the first have the `find_spec` method (that should be used instead of deprecated `find_module`) the latter don't.
The `FileFinder` objects come from the `sys.path_importer_cache`, in the method `_precache_zipimporters`, while the `ZipFInder` objects are added in the same method.
The `sys` module documentation explains that the objects in `sys.path_importer_cache` can be of two types, both of them having the `find_spec` method so i think it is ok to think of them as the same `FileFinder` type.
I wonder if it wouldn't be clearer to handle a container dedicated to `FileFinder` objects and another to `ZipFinder` objects, to distinguish clearly both types.
What do you think about it?

Reply:
Hm, as far as I see, the `sys.path_importer_cache` objects are all subclasses of `importlib.abc.Finder` (which is deprecated though) while `ZipFinder` is a class defined in Astroid and is a subclass of `Finder` which is defined in `interpreter/_import/spec.py`. We could certainly do an `isinstance` check instead of checking for the existence of the attribute, so this would become

```suggestion
            if isinstance(importer, (importlib.abc.MetaPathFinder, importlib.abc.PathEntryFinder)):
                found = importer.find_spec(modpath[0])
            elif isinstance(importer, spec.Finder):  # alternatively, we could use spec.ZipFinder
                found = importer.find_module(modpath[0])
```
and then we could do the same below. It is more explicit but less elegant in my opinion. Alternatively, we could modify `ZipFinder` (or perhaps `Finder`?) to expose a `find_spec` method which just refers to `find_module`, so the interfaces are better aligned and then we can just replace `find_module` with `find_spec` in this function.

Either solution works, but perhaps the last one is the most elegant one. What do you think?

Reply:
@GergelyKalmar i'm not fond either with the `isinstance` check.
By the way, i'm sorry because i made a mistake in my previous post. We are not dealing here with `ZipFinder` but with `zipimport.zipimporter` objects. Thus the second proposition you made doesn't hold.
Maybe the simplest thing to do is to let the `getattr` solution by explicitly commenting it to explain the two kinds of objects inside `pic.items()`...

Reply:
I agree – I think this is the most appropriate solution for now. I've added a comment explaining the situation and linking to this discussion for future reference accordingly. Thank you for your feedback!